### PR TITLE
Add more doc comments to pnp APIs

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: iothub/device/src/Convention/ClientPropertyCollection.cs [hunk 1]
```diff
@@ -251,6 +251,10 @@ public bool Contains(string componentName, string propertyName)
         /// <summary>
         /// Gets the value of a root-level property.
         /// </summary>
+        /// <remarks>
+        /// When retrieveing a writable client property acknowledgement payload <typeparamref name="T"/> should be
```
Comment (brycewang-microsoft) [inline iothub/device/src/Convention/ClientPropertyCollection.cs:255]:
```suggestion
        /// When retrieving a writable client property, acknowledgement payload <typeparamref name="T"/> should be
```
```suggestion
        /// When retrieveing a writable client property acknowledgement payload <typeparamref name="T"/> should be
```

File: iothub/device/src/Convention/ClientPropertyCollection.cs [hunk 2]
```diff
@@ -273,6 +277,10 @@ public bool TryGetValue<T>(string propertyName, out T propertyValue)
         /// <summary>
         /// Gets the value of a component-level property.
         /// </summary>
+        /// <remarks>
+        /// When retrieveing a writable client property acknowledgement payload <typeparamref name="T"/> should be
```
Comment (brycewang-microsoft) [inline iothub/device/src/Convention/ClientPropertyCollection.cs:281]:
```suggestion
        /// When retrieving a writable client property, acknowledgement payload <typeparamref name="T"/> should be
```

File: iothub/device/src/Convention/WritableClientPropertyCollection.cs [hunk 3]
```diff
@@ -86,15 +86,20 @@ public bool Contains(string componentName, string propertyName)
...
+        /// Gets the value of a root-level property as a <see cref="WritableClientProperty"/>.
         /// </summary>
+        /// <remarks>
+        /// <see cref="WritableClientProperty"/> has a convenience method to help you build the writable property acknowledgement object.
+        /// To retrieve the value of the root-level writable property update request see <see cref="TryGetValue{T}(string, out T)"/>
+        /// or <see cref="WritableClientProperty.TryGetValue{T}(out T)"/>.
+        /// </remarks>
         /// <param name="propertyName">The property to get.</param>
-        /// <param name="propertyValue">When this method returns true, this contains the value of the root-level property.
+        /// <param name="writableClientProperty">When this method returns true, this contains the value of the root-level property.
```
Comment (brycewang-microsoft) [inline iothub/device/src/Convention/WritableClientPropertyCollection.cs:97]:
```suggestion
        /// <param name="writableClientProperty"> When this method returns true, this contains the value of the root-level property.
```
```suggestion
        /// <param name="writableClientProperty">When this method returns true, this contains the value of the root-level property.
```

Comment (abhipsaMisra) [reply iothub/device/src/Convention/WritableClientPropertyCollection.cs:97]: We don't add whitespace between the param name and the description so I'm going to leave it as-is.

File: iothub/device/src/Convention/WritableClientPropertyCollection.cs [hunk 4]
```diff
@@ -141,16 +148,21 @@ public bool TryGetValue<T>(string propertyName, out T propertyValue)
...
         /// </summary>
+        /// <remarks>
+        /// <see cref="WritableClientProperty"/> has a convenience method to help you build the writable property acknowledgement object.
+        /// To retrieve the value of the component-level writable property update request see <see cref="TryGetValue{T}(string, string, out T)"/>
+        /// or <see cref="WritableClientProperty.TryGetValue{T}(out T)"/>.
+        /// </remarks>
         /// <param name="componentName">The component which holds the required property.</param>
         /// <param name="propertyName">The property to get.</param>
-        /// <param name="propertyValue">When this method returns true, this contains the value of the component-level property.
+        /// <param name="writableClientProperty">When this method returns true, this contains the value of the component-level property.
```
[the remark this answers is not in the document]
Comment (abhipsaMisra) [reply iothub/device/src/Convention/WritableClientPropertyCollection.cs:160]: We don't add whitespace between the param name and the description so I'm going to leave it as-is.

**Pull request merged**